garden_ai/client.py

```diff
@@ -346,13 +346,24 @@ def _mint_draft_doi(self, test: bool = True) -> str:
     def _update_datacite(self, obj: Union[Garden, Pipeline]) -> None:
         logger.info("Requesting update to DOI")
         url = f"{GARDEN_ENDPOINT}/doi"
+        header = {"Authorization": self.garden_authorizer.get_authorization_header()}
+
+        metadata = json.loads(
+            obj.json() if isinstance(obj, Pipeline) else obj.expanded_json()
+        )
+        metadata.update(
+            creators=[{"name": name} for name in metadata["authors"]],
+            titles=[{"title": metadata["title"]}],
+            publicationYear=metadata["year"],
+            publisher="Garden AI",
+            types={"resourceTypeGeneral": "Text"},
+            event="publish",
+            url=f"https://thegardens.ai/{obj.doi}",
+        )
+        del metadata["authors"]
+        del metadata["title"]
+        del metadata["year"]
```

Comment:
The logic here looks good, thanks for figuring out what DataCite _actually_ wants.

Taking a closer look at this, and some of our existing code, could you implement this data transformation as `datacite_json` methods on the Pipeline and Garden classes? It looks like Owen had made some datacite-specific formatting logic that we would want to replace/amend if it isn't right. And that way we can add some easy unit tests that 
1. create a Garden and test that its datacite_json is as expected
2. create a Pipeline and test that its datacite_json is as expected

Reply:
(And just in terms of old-school OOP fundamentals we keep the data transformation next to the data instead of mixing it in with this external API invocation method)

Reply:
I am not 100% sure I'm picking up what you're putting down, but I refactored the fix using that method and added a couple tests.